hummingbot/client/hummingbot_application.py

```diff
@@ -1141,6 +1142,99 @@ def compare_balance_snapshots(self):
+        :param quote: tuple that is the quote currency token pair
+        """
+        market_name = market_symbol_pair.market.name
+
+        if is_base:
+            asset_tuple = base
+            asset = market_symbol_pair.base_asset
+        else:
+            asset_tuple = quote
+            asset = market_symbol_pair.quote_asset
+
+        if is_starting:
+            amount = self.starting_balances.get(asset).get(market_name)
+        else:
+            amount = self.balance_snapshot().get(asset).get(market_name)
+
+        asset_list = list(asset_tuple)
+        if len(asset_list) == 0:
+            asset_list.insert(0, asset)
+            asset_list.insert(1, float(amount))
+        else:
+            if asset_list[0] == asset:
+                asset_list[1] += float(amount)
+            else:
+                erc = ExchangeRateConversion.get_instance()
+                asset_list[1] += float(erc.convert_token_value(
+                    amount=float(amount),
+                    from_currency=asset,
+                    to_currency=asset_list[0]
+                ))
+
+        asset_tuple = tuple(asset_list)
+        if is_base:
+            base = asset_tuple
+        else:
+            quote = asset_tuple
+
+        return base, quote
+
+    def analyze_performance(self):
+        """ Determine the profitability of the trading bot. """
+        if len(self.starting_balances) == 0:
+            self._notify("  Performance analysis is not available before bot starts")
+            return
+
+        # Each of these is in the format (currency, amount)
+        starting_base = tuple()
+        starting_quote = tuple()
+        current_base = tuple()
+        current_quote = tuple()
+
+        for market_symbol_pair in self.market_symbol_pairs:
+            for is_base in [True, False]:
+                starting_base, starting_quote = self.add_balances(
+                    market_symbol_pair=market_symbol_pair,
+                    is_base=is_base,
+                    is_starting=True,
+                    base=starting_base,
+                    quote=starting_quote
+                )
+                current_base, current_quote = self.add_balances(
+                    market_symbol_pair=market_symbol_pair,
+                    is_base=is_base,
+                    is_starting=False,
+                    base=current_base,
+                    quote=current_quote
+                )
+
+        # Compute the current exchange rate. We use the first market_symbol_pair because
+        # if the trading pairs are different, such as WETH-DAI and ETH-USD, the tuples
+        # above will contain the information in terms of the first trading pair based on
+        # the way that the balances were computed in the for loop above.
+        market_pair_info = self.market_symbol_pairs[0]
+        market = market_pair_info.market
+        buy_price = market.get_price(symbol=market_pair_info.trading_pair, is_buy=True)
```

Comment:
Again, don't add unnecessary keywords. It actually makes the code more fragile than without.